### PR TITLE
Create Errata Class 

### PR DESCRIPTION
**Pull request by rpledger14**
- Create overall Errata module for hardware errata and their respective software workarounds
- This module includes HwErratum and SwErratumWorkaround classes
- Add spec test for Errata

### CODE REVIEW

File: lib/origen/errata.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    end
+ 
+    def _sw_workarounds
+      @_sw_workarounds ||= Hash.new do |h,k|
+        h[k] = {}
+      end
+    end
+
+    # Return a hash based on the filter provided
+    def filter_hash(hash, filter)
```
Comment (welguisz) [inline lib/origen/errata.rb]: Remove this function.  For some reason this function is clashing with the same function name in specs.rb.  When I removed this, the 'origen specs' command passed.

File: lib/origen/errata.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+    # Returns a sw_workaround object with a specific id
+    def sw_workarounds(options = {})
+      options = {
+        id: nil
+      }.update(options)
+      return nil if @_sw_workarounds.nil?
+      return nil if @_sw_workarounds.empty?
+
+      sw_workarounds_found = Hash.new do |h, k|
```
Comment (welguisz) [inline lib/origen/errata.rb]: This should be:
sw_workarounds_found = {}
since this is a 1-D hash.

File: lib/origen/errata.rb [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    def _errata
+      @_errata ||= Hash.new do |h,k|
+        h[k] = Hash.new do |hh, kk|
+          hh[kk] = {}
+         end
+      end
+    end
+ 
+    def _sw_workarounds
+      @_sw_workarounds ||= Hash.new do |h,k|
```
Comment (welguisz) [inline lib/origen/errata.rb]: Should be \@sw_workarounds = {} since it is a 1-D array

File: spec/errata_spec.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @dut.errata(id: "ERR1").sw_workaround.description.should == "This is how Feature A is fixed"
+    @dut.errata(id: "ERR1").sw_workaround.sw_disposition.should == "Available"
+    @dut.errata(id: "ERR1").sw_workaround.distribution.should == "SDK1"
+    @dut.errata(id: "ERR1").sw_workaround.note.should == "Release note for workaround"
+    @dut.errata(id: "ERR1").sw_workaround.patches.should == "www.patch.com/1"
+  end
+
+  it "Errata function returns object with specific ID" do
+    @dut.errata(id: "ERR1").is_a?(Origen::Errata::HwErratum).should == true
+    @dut.errata(id:"ERR1").id.should == "ERR1"
```
Comment (welguisz) [inline spec/errata_spec.rb]: There should be a space after **:**.
Line should be:

`   @dut.errata(id: 'ERR1').id.should == 'ERR1'
`

File: lib/origen/errata.rb [hunk 5]
```diff
@@ -92,9 +92,7 @@ def _errata
     end
  
     def _sw_workarounds
-      @_sw_workarounds ||= Hash.new do |h,k|
```
Comment (welguisz) [inline lib/origen/errata.rb]: change this back to being a 2-D hash.  For some reason, the 1-D hash is causing origen specs to fail.

**Comment by welguisz** [comment]
Can you run 'origen lint' and make any changes needed by it.

**Comment by welguisz** [comment]
@ginty This looks ready to be merged and a new release for Origen soon.  Thanks!!!

**Comment by priyavadan** [comment]
@ginty @welguisz @rpledger14 Pull Request #48 has been merged.

**Pull request merged**